go-controller/pkg/ovn/egressip_test.go

```diff
@@ -3410,6 +3410,167 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations cluster default network"
 		)
 	})
 
+	ginkgo.Context("IPv4 on pod UPDATE", func() {
+		ginkgo.It("does not reconfigure or remove existing pod config if no change", func() {
+			config.OVNKubernetesFeature.EnableInterconnect = true
+			app.Action = func(*cli.Context) error {
+				egressPod := *newPodWithLabels(eipNamespace, podName, node1Name, podV4IP, egressPodLabel)
+				egressNamespace := newNamespace(eipNamespace)
+				nodeIPv4 := "192.168.126.210/24"
+				egressIP := net.ParseIP("192.168.126.211")
+				_, nodeSubnetV4, _ := net.ParseCIDR(v4Node1Subnet)
+				_, nodeSubnetV6, _ := net.ParseCIDR(v6Node1Subnet)
+
+				annotations := map[string]string{
+					"k8s.ovn.org/node-primary-ifaddr":             fmt.Sprintf("{\"ipv4\": \"%s\", \"ipv6\": \"%s\"}", nodeIPv4, ""),
+					"k8s.ovn.org/node-subnets":                    fmt.Sprintf("{\"default\":\"%s\",\"%s\"}", v4Node1Subnet, v6Node1Subnet),
+					"k8s.ovn.org/node-transit-switch-port-ifaddr": "{\"ipv4\":\"100.88.0.2/16\", \"ipv6\": \"fd97::2/64\"}",
+					util.OVNNodeHostCIDRs:                         fmt.Sprintf("[\"%s\"]", nodeIPv4),
+					"k8s.ovn.org/zone-name":                       node1Name,
+				}
+				node := getNodeObj(node1Name, annotations, map[string]string{}) // add node to avoid errori-ing out on transit switch IP fetch
+				fakeOvn.startWithDBSetup(
+					libovsdbtest.TestSetup{
+						NBData: []libovsdbtest.TestData{
+							&nbdb.LogicalRouterPort{
+								UUID:     types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + node1Name + "-UUID",
+								Name:     types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + node1Name,
+								Networks: []string{nodeLogicalRouterIfAddrV6, nodeLogicalRouterIfAddrV4},
+							},
+							&nbdb.LogicalRouter{
+								Name: types.OVNClusterRouter,
+								UUID: types.OVNClusterRouter + "-UUID",
+							},
+							&nbdb.LogicalRouter{
+								Name:    types.GWRouterPrefix + node1Name,
+								UUID:    types.GWRouterPrefix + node1Name + "-UUID",
+								Ports:   []string{types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + node1Name + "-UUID"},
+								Options: map[string]string{"dynamic_neigh_routers": "false"},
+							},
+							&nbdb.LogicalSwitchPort{
+								UUID: "k8s-" + node.Name + "-UUID",
+								Name: "k8s-" + node.Name,
+								Addresses: []string{"fe:1a:b2:3f:0e:fb " + util.GetNodeManagementIfAddr(nodeSubnetV4).IP.String(),
+									"fe:1a:b2:3f:0e:fb " + util.GetNodeManagementIfAddr(nodeSubnetV6).IP.String()},
+							},
+							&nbdb.LogicalSwitch{
+								UUID:  node.Name + "-UUID",
+								Name:  node.Name,
+								Ports: []string{"k8s-" + node.Name + "-UUID"},
+							},
+						},
```

Comment:
can we maybe move this to a parametrized helper ? 

I find it hard to review like this. Not a must, but I think it'll help w/ the test readability.